src/axom/inlet/docs/sphinx/simple_types.rst

```diff
@@ -0,0 +1,144 @@
+############
+Simple Types
+############
+
+To help structure your input file, Inlet categorizes the information into
+two types: Fields and Tables.
+
+Fields refer to the individual scalar values that are either at the global level or
+that are contained inside of a Table.
+
+Tables can contain multiple Fields, other sub-Tables, as well as a single array or a single dictionary.
+
+.. note::  There is an inherent global Table that holds all global level Fields.  This can be
+  accessed via your `Inlet` class instance.
+
+
+*******
+Fields
+*******
+
+In Inlet, Fields represent an individual scalar value of primitive type.
+There are four supported field types: ``bool``, ``int``, ``double``, and ``string``.
+
+In this example we will be using the following part of an input file:
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_input_start
+   :end-before: _inlet_simple_types_fields_input_end
+   :language: lua
+
+
+Defining And Storing
+--------------------
+
+This example shows how to add the four simple field types with descriptions to the
+input file schema and add their values, if present in the input file, to the Sidre
+DataStore to be accessed later.
+
+.. literalinclude:: ../../examples/fields.cpp
```

Comment:
The code snippet here might benefit from some comments. E.g.

```cpp
  // register a boolean with inlet's (associated with its top-level table)
  inlet.addBool("a_simple_bool", "A description of a_simple_bool");

  // register an int with inlet
  inlet.addInt("a_simple_int", "A description of a_simple_int");

  // register a double with inlet and mark it as required
  inlet.addDouble("a_simple_double", "A description of a_simple_double").required();

  // register a string with inlet
  inlet.addString("a_simple_string", "A description of a_simple_string");

  // register an int with inlet and provide a default value
  inlet.addInt("a_defaulted_int", "An int that has a default value").defaultValue(17);

  // register a string with inlet
  inlet.addString("does_not_exist", "Should not be in your input file");
```

Note: I don't understand the comment about `does_not_exist` not being in the input. It seems a bit confusing. What would happen if it is there.